example/createCounter.js

```diff
@@ -36,38 +36,42 @@ export default function (rootElem: ?HTMLElement): Counter {
 	_decrementButton.addEventListener('click', _handleDecrementButtonClick);
 	_incrementButton.addEventListener('click', _handleIncrementButtonClick);
 
-	function _checkValue(expected) {
-		expect(Number(expected)).toEqual(_value);
-		return _value;
-	}
-
 	let _fakeSuccessValue = false;
 	let _fakeErrorValue = false;
 
 	clientOpened.then(() => {
 		// simple regression steps
-		client.publishStep('.* call a step with a result', () => 'hello!');
-		client.publishStep('.* call a step without a result', () => {});
-		client.publishStep('.* call a step which fails', () => { expect(true).toBe(false); });
-		client.publishStep('.* call a step which succeeds asynchronously', () => {
+		client.publishStep('.*call a step with a result', () => 'hello!');
+		client.publishStep('.*call a step without a result', () => {});
+		client.publishStep('.*call a step which fails', () => { expect(true).toBe(false); });
+		client.publishStep('.*call a step which succeeds asynchronously', () => {
 			setTimeout(() => { _fakeSuccessValue = true; }, 500);
 			expect(_fakeSuccessValue).toBe(true);
 			return String(_fakeSuccessValue);
 		}, {
 			retryDuration: 1000,
 		});
-		client.publishStep('.* call a step which times out', () => {
+		client.publishStep('.*call a step which times out', () => {
 			setTimeout(() => { _fakeErrorValue = true; }, 1000);
 			expect(_fakeErrorValue).toBe(true);
 			return String(_fakeErrorValue);
 		}, {
 			retryDuration: 500,
 		});
+		client.publishStep('.*in chorus-js \'(.*)\' has the value \'(.*)\'', ([name, value], context) => {
```

Comment:
The function simply receives as the first argument the arguments sent from chorus as an array. The syntax you see here is part of ES6 and is called `destructuring`. I'm `"picking"` out the 2 first values out of the array and naming them right there in the function parameters.

I could have done this instead:

```
client.publishStep('.*in chorus-js \'(.*)\' has the value \'(.*)\'', (args, context) => {
    const [name, value] = args;
    console.log(name, value); // I now have access to these directly
    ....
```